### PR TITLE
Use Vendetta for dependency management

### PR DESCRIPTION
**Pull request by lclc**
Vendetta is a minimal tool for managing the dependencies in the vendor directory of a go project.
It adds all Go packages as git submodules, to make sure updates of Go packages don't break the build.

See: https://github.com/dpw/vendetta

Built on top of https://github.com/joushou/sshmuxd/pull/14, but I can also do it without the dependencies from there.

### CODE REVIEW

File: README.md
Comment (kennylevinsen) [inline README.md]: If you do the following, you don't need "mkdir", a "cd", or the submodule update you added later:
```shell
git clone --recursive https://github.com/joushou/sshmuxd.git $GOPATH/src/github.com/joushou/sshmuxd
```

File: README.md
Comment (kennylevinsen) [inline README.md]: If you add `--recursive` to the clone, this becomes unnecessary.

File: README.md
Comment (kennylevinsen) [inline README.md]: go build is implicit in go install - maybe just let that be the only step?

File: .gitmodules
Comment (kennylevinsen) [inline .gitmodules:48]: Damn, that's a long list. I guess this is pulled in by viper? Not complaining, just... Damn.

**Review by kennylevinsen — CHANGES_REQUESTED** [review]
Just a minor comment on the README. Will also merge the other PR first, as this contains both commits.

File: README.md
Comment (kennylevinsen) [reply README.md]: Note that I think the "cd to final sshmuxd folder" kinda belongs down in the build/install section.

**Comment by kennylevinsen** [comment]
I'm entirely fine with using vendored submodules. I didn't previously because my dependencies where quite minimal (and I was lazy at the time - work and all). As long as the tool is merely an external convenience that users/developers won't *need* to use, then I'm all okay with it.

**Comment by lclc** [comment]
Thanks for the feedback, I've changed the README accordingly.

I plan to update & rebase this PR again after the other gets merged.

**Comment by kennylevinsen** [comment]
Cool, I'll merge after the other one. Git should be smart enough to not need a rebase.

**Review by kennylevinsen — APPROVED** [review]

**Pull request merged**
